python/ray/utils.py

```diff
@@ -496,6 +497,141 @@ def is_main_thread():
     return threading.current_thread().getName() == "MainThread"
 
 
+def set_psigdeath(child_proc, _globals=[None]):
+    """Ensures the child process dies with the parent (fate-sharing).
+
+    On Linux, this must be called by the child in preexec_fn.
```

Comment:
Can we separate linux & windows into two separate functions? I find it much harder to follow when we have the windows/linux checks littered throughout the code. Would also avoid having this `child_proc` argument that has different semantics depending on platform...

Reply:
Also having to call this from different processes on linux vs windows is a bit wonky

Reply:
Yeah sure, I can split it into two functions. I can't change how it's called though... that's just a function of what the APIs provide.